source/beamline/BioXAS/BioXASSSRLMonochromatorRegionControl.cpp

```diff
@@ -0,0 +1,846 @@
+{
+	if (!brakeStatus_ && brakeStatus_ != brakeStatus) {
+		brakeStatus_ = brakeStatus;
+		addChildControl(brakeStatus_);
+
+		connect( brakeStatus_, SIGNAL(valueChanged(double)), this, SIGNAL(brakeStatusChanged(double)) );
+		connect( brakeStatus_, SIGNAL(movingChanged(bool)), this, SIGNAL(movingChanged(bool)) );
+	}
+}
+
+void BioXASSSRLMonochromatorRegionControl::setBraggControl(AMControl *bragg)
+{
+	if (!bragg_ && bragg_ != bragg) {
+		bragg_ = bragg;
+		addChildControl(bragg_);
+
+		connect( bragg_, SIGNAL(movingChanged(bool)), this, SIGNAL(movingChanged(bool)) );
+	}
+}
+
+void BioXASSSRLMonochromatorRegionControl::setBraggAtCrystalChangePositionStatusControl(AMControl *inPosition)
+{
+	if (!braggAtCrystalChangePositionStatus_ && braggAtCrystalChangePositionStatus_ != inPosition) {
+		braggAtCrystalChangePositionStatus_ = inPosition;
+		addChildControl(braggAtCrystalChangePositionStatus_);
+
+		connect( braggAtCrystalChangePositionStatus_, SIGNAL(valueChanged(double)), this, SIGNAL(braggAtCrystalChangePositionStatusChanged(double)) );
+		connect( braggAtCrystalChangePositionStatus_, SIGNAL(movingChanged(bool)), this, SIGNAL(movingChanged(bool)) );
+	}
+}
+
+void BioXASSSRLMonochromatorRegionControl::setCrystalChangeControl(AMControl *crystalChange)
+{
+	if (!crystalChange_ && crystalChange_ != crystalChange) {
+		crystalChange_ = crystalChange;
+		addChildControl(crystalChange_);
+
+		connect( crystalChange_, SIGNAL(movingChanged(bool)), this, SIGNAL(movingChanged(bool)) );
+	}
+}
+
+void BioXASSSRLMonochromatorRegionControl::setCrystalChangeCWLimitStatusControl(AMControl *limitStatus)
+{
+	if (!crystalChangeCWLimitStatus_ && crystalChangeCWLimitStatus_ != limitStatus) {
+		crystalChangeCWLimitStatus_ = limitStatus;
+		addChildControl(crystalChangeCWLimitStatus_);
+
+		connect( crystalChangeCWLimitStatus_, SIGNAL(movingChanged(bool)), this, SIGNAL(movingChanged(bool)) );
+	}
+}
+
+void BioXASSSRLMonochromatorRegionControl::setCrystalChangeCCWLimitStatusControl(AMControl *limitStatus)
+{
+	if (!crystalChangeCCWLimitStatus_ && crystalChangeCCWLimitStatus_ != limitStatus) {
+		crystalChangeCCWLimitStatus_ = limitStatus;
+		addChildControl(crystalChangeCWLimitStatus_);
+
+		connect( crystalChangeCCWLimitStatus_, SIGNAL(movingChanged(bool)), this, SIGNAL(movingChanged(bool)) );
+	}
+
+}
+
+void BioXASSSRLMonochromatorRegionControl::setRegionAStatusControl(AMControl *regionStatus)
+{
+	if (!regionAStatus_ && regionAStatus_ != regionStatus) {
+		regionAStatus_ = regionStatus;
+		addChildControl(regionAStatus_);
+
+		connect( regionAStatus_, SIGNAL(valueChanged(double)), this, SLOT(onRegionControlValueChanged()) );
+		connect( regionAStatus_, SIGNAL(movingChanged(bool)), this, SIGNAL(movingChanged(bool)) );
+	}
+}
+
+void BioXASSSRLMonochromatorRegionControl::setRegionBStatusControl(AMControl *regionStatus)
+{
+	if (!regionBStatus_ && regionBStatus_ != regionStatus) {
```

Comment:
You are completely correct that the code - as currently written - is safe.  They are only called in the constructor and for the time being that is all that matters.  However, there is a case where a subclass could do ```setXYZControl(0)``` which would cause all kinds of issues.  

I wouldn't worry about it too much at the moment.  Maybe just flag it for code cleanup at a later date?